key-manager-oapi.yaml

```diff
@@ -0,0 +1,225 @@
+                            - error: Any other status different to the above: decrypting error, I/O errors, etc.
+                          enum:
+                            - imported
+                            - duplicate
+                            - error
+                        message:
+                          type: string
+                          description: error message if status == error
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "500":
+          $ref: "#/components/responses/InternalError"
+
+    delete:
+      operationId: DeleteKeys
+      summary: Delete Keys.
+      description: |
+        DELETE must delete all keys from `request.pubkeys` that are known to the keymanager and exist in its
+        persistent storage. Additionally, DELETE must fetch the slashing protection data for the requested keys from
+        persistent storage, which must be retained (and not deleted) after the response has been sent. Therefore in the
+        case of two identical delete requests being made, both will have access to slashing protection data.
+
+        In a single atomic sequential operation the keymanager must:
+        1. Guarantee that key(s) can not produce any more signature; only then
+        2. Delete key(s) and serialize its associated slashing protection data
+
+        DELETE should never return a 404 response, even if all pubkeys from request.pubkeys have no extant keystores
+        nor slashing protection data.
+      security:
+        - cookieAuth: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                pubkeys:
+                  type: array
+                  description: List of public keys to delete.
+                  items:
+                    $ref: "#/components/types/Pubkey"
+      responses:
+        "200":
+          description: Success response
+          content:
+            application/json:
+              schema:
+                title: DeleteKeysResponse
+                type: object
+                properties:
+                  data:
+                    type: array
+                    description: Deletion status of all keys in `request.pubkeys` in the same order.
+                      type: object
+                      properties:
+                        status:
+                          type: string
+                          description: |
+                            - deleted: Successfully deleted from persistent storage and not able to sign any messages
+                            - not-found: Pubkey not known to keymanager
+                            - error: Any other status different to the above: I/O errors, etc.
+                          enum:
+                            - deleted
+                            - not-found
```

Comment:
not sure about other languages, but 'not-found' isn't enum friendly at least in java - could we have `not_found` or something?

For the scenario where the key is already removed and we're just giving slashing protection data back, might be worth a value so we can use other than the error conditions

```
deleted: key was active and removed
not_active: slashing protection data returned but key was not active
not_found: key was not found to be removed, and no slashing data can be returned
error: unexpected condition meant the key could not be removed (the key was actually found, but we couldn't stop using it) - this would be a sign that making it active elsewhere would almost certainly cause you headaches / slashing conditions etc.
```

Reply:
As discussed on Discord, I'm on-board with the above. Underscore naming is also more consistent with the beacon APIs (e.g. the validator statuses)